sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -135,6 +135,34 @@ abstract class JdbcDialect extends Serializable with Logging {
     s""""$colName""""
   }
 
+  /**
+   * Create the table if the table does not exist.
+   * To allow certain options to append when create a new table, which can be
+   * table_options or partition_options.
+   * E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+   * @param statement
+   * @param tableName
+   * @param strSchema
+   * @param options
+   */
+  def createTable(
+      statement: Statement,
+      tableName: String,
+      strSchema: String,
+      options: JdbcOptionsInWrite): Unit = {
+    val createTableOptions = options.createTableOptions
+    statement.executeUpdate(s"CREATE TABLE $tableName ($strSchema) $createTableOptions")
+    if (options.tableComment.nonEmpty) {
+      try {
+        val tableCommentQuery = getTableCommentQuery(tableName, options.tableComment)
+        statement.executeUpdate(tableCommentQuery)
```

Comment:
to confirm: is it safe to ignore the error? Now we use a single `Statement` to create table and add comment.

Reply:
In the `createTable` of `JdbcUtils`, there is a finally code block to close the `Statement`

Reply:
Create table and add comment share the same statement here. It will lead to create table failed when adding comment occurs error if statement not be committed.

Reply:
`Statement` is automatically committed by default, and it will be committed after executing a SQL statement. Therefore, there is no case where the commit of create table fails due to the exception of add comment. A single `Statement` is also used multiple times in `createSchema`, I don't think there is a problem

Reply:
Oh, I forgot it.